examples/Spirit/spirit_utils.cc

```diff
@@ -750,79 +750,87 @@ double calcTorqueInt(
-
-        // ith power variables
-        drake::symbolic::Variable power_plus_i = power_pluses[power_pluses.size()-1];
-        drake::symbolic::Variable power_minus_i = power_minuses[power_minuses.size()-1];
-
-        // Get current actuation and state
-        auto u_i = trajopt.input(trajopt.get_mode_start(mode_index) + knot_index);
-        auto x_i   = trajopt.state_vars(mode_index, knot_index);
-        drake::symbolic::Variable actuation = u_i(actuator_map.at("motor_" + std::to_string(joint)));
-        drake::symbolic::Variable velocity = x_i(n_q + velocities_map.at("joint_" + std::to_string(joint) +"dot"));
-
-        // Constrain newly power variables
-        if (cost_work_gain > 0){
-          trajopt.AddConstraint((actuation * velocity + Q * actuation * actuation) * work_constraint_scale == (power_plus_i - power_minus_i) * work_constraint_scale) ;
-          trajopt.AddLinearConstraint(power_plus_i * work_constraint_scale >= 0);
-          trajopt.AddLinearConstraint(power_minus_i * work_constraint_scale >= 0);
-        }
-        trajopt.SetInitialGuess(power_plus_i, 0);
-        trajopt.SetInitialGuess(power_minus_i, 0);
+      for (int knot_index = 0; knot_index < trajopt.mode_length(mode_index)-1; knot_index++) {
+        auto u_i = trajopt.input(trajopt.get_mode_start(mode_index) + knot_index)(act_int);
+        auto x_i   = trajopt.state_vars(mode_index, knot_index)(vel_int);
+        auto u_ip = trajopt.input(trajopt.get_mode_start(mode_index) + knot_index+1)(act_int);
+        auto x_ip   = trajopt.state_vars(mode_index, knot_index+1)(vel_int);
+
+        auto hi = trajopt.timestep(trajopt.get_mode_start(mode_index) + knot_index);
+        drake::solvers::VectorXDecisionVariable variables(5);
+        variables(0) = hi[0];
+        variables(1) = u_i;
+        variables(2) = u_ip;
+        variables(3) = x_i;
+        variables(4) = x_ip;
+        cost_joint_work_bindings.push_back(trajopt.AddCost(joint_work_cost, variables));
+      } // knot point loop
+    } // Mode loop
+  } // Joint loop
+  return cost_joint_work_bindings;
+}
 
-        // For 0th iteration, dont bother adding cost
-        if(knot_index > 0){
+JointWorkCost::JointWorkCost(const drake::multibody::MultibodyPlant<double>& plant,
+                             const double &Q,
+                             const double &cost_work,
+                             const double &alpha,
+                             const std::string &description)
+    :solvers::NonlinearCost<double>(5,description), plant_(plant){
+  Q_ = Q;
+  cost_work_ = cost_work;
+  alpha_ = alpha;
+  n_q_ = plant_.num_positions();
+  n_v_ = plant_.num_velocities();
+  n_u_ = plant_.num_actuators();
+  DRAKE_DEMAND(alpha_ > 0);
+}
 
-          auto u_im = trajopt.input(trajopt.get_mode_start(mode_index) + knot_index-1);
-          drake::symbolic::Variable actuation_m = u_im(actuator_map.at("motor_" + std::to_string(joint)));
+double JointWorkCost::relu(const double x) const{
```

Comment:
might want to put smooth or approx in the function name since this not technically a relu right?